WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -98,75 +106,105 @@ public int onStartCommand(Intent intent, int flags, int startId) {
+            PostModel post = mPostStore.getPostByLocalPostId(media.getPostId());
+            if (post != null) {
+                String content = post.getContent();
+                if (content != null) {
+
+                    // actually replace the media ID with the media uri
+                    MediaUploadReadyListener processor = new MediaUploadReadyProcessor();
+                    PostModel modifiedPost = processor.replaceMediaFileWithUrlInPost(post, String.valueOf(media.getId()), FluxCUtils.mediaFileFromMediaModel(media));
+                    if (modifiedPost != null) {
+                        post = modifiedPost;
+                    }
+
+                    // we changed the post, so let’s mark this down
+                    if (!post.isLocalDraft()) {
+                        post.setIsLocallyChanged(true);
+                    }
+                    post.setDateLocallyChanged(DateTimeUtils.iso8601FromTimestamp(System.currentTimeMillis() / 1000));
+
+                    // finally save the post
+                    savePostToDb(post);
+                }
+            }
         }
+    }
 
-        // somehow lost our reference to the site, complete this action
-        if (mSite == null) {
-            AppLog.i(AppLog.T.MEDIA, "Unexpected state, site is null. Skipping this request - MediaUploadService.");
+    private synchronized void savePostToDb(PostModel post) {
+        mDispatcher.dispatch(PostActionBuilder.newUpdatePostAction(post));
+    }
+
+    private void uploadNextInQueue() {
+
+        MediaModel next = getNextMediaToUpload();
+
+        if (next == null) {
+            AppLog.v(AppLog.T.MEDIA, "No more media items to upload. Skipping this request - MediaUploadService.");
             stopServiceIfUploadsComplete();
             return;
         }
 
-        mCurrentUpload = getNextMediaToUpload();
+        SiteModel site = mSiteStore.getSiteByLocalId(next.getLocalSiteId());
 
-        if (mCurrentUpload == null) {
-            AppLog.v(AppLog.T.MEDIA, "No more media items to upload. Skipping this request - MediaUploadService.");
+        // somehow lost our reference to the site, complete this action
+        if (site == null) {
+            AppLog.i(AppLog.T.MEDIA, "Unexpected state, site is null. Skipping this request - MediaUploadService.");
             stopServiceIfUploadsComplete();
-            return;
         }
 
-        dispatchUploadAction(mCurrentUpload);
+        dispatchUploadAction(next, site);
     }
 
-    private void completeCurrentUpload() {
-        if (mCurrentUpload != null) {
-            getCompletedItems().add(mCurrentUpload);
-            getUploadQueue().remove(mCurrentUpload);
-            mCurrentUpload = null;
+    private void completeUploadWithId(int id) {
+        getUploadQueue().remove(getMediaFromQueueById(id));
+        stopServiceIfUploadsComplete();
+    }
+
+    private MediaModel getMediaFromQueueById(int id) {
+        if (mQueue != null && mQueue.size() > 0) {
```

Comment:
Could we use `getUploadQueue()` here instead of `mQueue` directly for consistency (like in `getNextMediaToUpload()` below)?